src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web.UnitTests/Controllers/ManageApprenticesTests/WhenIGoToManageMyApprentices.cs

```diff
@@ -1,42 +1,105 @@
-﻿using Microsoft.AspNetCore.Mvc;
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoFixture;
+using AutoFixture.NUnit3;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
 using NUnit.Framework;
+using SFA.DAS.Commitments.Shared.Interfaces;
+using SFA.DAS.CommitmentsV2.Types;
 using SFA.DAS.ProviderCommitments.Web.Controllers;
 using SFA.DAS.ProviderCommitments.Web.Models;
+using SFA.DAS.Testing.AutoFixture;
 
 namespace SFA.DAS.ProviderCommitments.Web.UnitTests.Controllers.ManageApprenticesTests
 {
     [TestFixture]
     public class WhenIGoToManageMyApprentices
     {
-        private ManageApprenticesController _controller;
+        [Test, MoqAutoData]
+        public void IfCalledWithAnInvalidRequestShouldGetBadResponseReturned(
+            [Frozen]Mock<ICommitmentsService> commitmentsService,
+            ManageApprenticesController controller)
+        {
+            //Arrange
+            controller.ModelState.AddModelError("test", "test");
+
+            //Act
+            var result = controller.Index(1);
+
+            //Assert
+            Assert.IsInstanceOf<BadRequestObjectResult>(result.Result);
+        }
 
-        [SetUp]
-        public void Arrange()
+        [Test, MoqAutoData]
+        public void ThenTheProviderIdIsPassedToTheViewModel(
+            [Frozen]Mock<ICommitmentsService> commitmentsService,
+            ManageApprenticesController controller,
+            int providerId)
         {
-            _controller = new ManageApprenticesController();
+            //Arrange
+            
+            //Act
+            var result = controller.Index(providerId);
+
+            var view = ((ViewResult) result.Result).Model as ManageApprenticesViewModel;
+
+            //Assert
+            Assert.AreEqual(view.ProviderId, providerId);
         }
 
-        [Test]
-        public void IfCalledWithAnInvalidRequestShouldGetBadResponseReturned()
+        [Test, MoqAutoData]
+        public void ThenTheApprovedApprenticesAreSet(
+            [Frozen]Mock<ICommitmentsService> commitmentsService,
+            ManageApprenticesController controller,
+            List<ApprenticeshipDetails> approvedApprenticeships)
         {
             //Arrange
-            _controller.ModelState.AddModelError("test", "test");
+            commitmentsService.Setup(x => x.GetApprovedApprenticeships(It.IsAny<uint>()))
+                .ReturnsAsync(approvedApprenticeships);
+
             //Act
-            var result = _controller.Index(1);
+            var result = controller.Index(1);
+            var view = ((ViewResult)result.Result).Model as ManageApprenticesViewModel;
+
             //Assert
-            Assert.IsInstanceOf<BadRequestObjectResult>(result);
+            Assert.AreEqual(view.Apprenticeships.Count(), approvedApprenticeships.Count);
```

Comment:
I'd check more than just the count as the details being sent over may not be mapped correctly etc. The unit test needs to make sure the details that we are sent reach the view correctly. This will then catch if the models being sent to the view haven't be corrupted accidentally

Reply:
Although the updated test code is in the right direction it only checks the first instance when it should check all of them just in case the mapping doesn't get applied to all. You can normally do this in a single line check using fluent assertion. Have you used obj.Should().BeEquivalentTo(obj2) (https://fluentassertions.com/objectgraphs/) before? If not I can show you how it works and can solve this issue.